unlock-js/src/web3Service.js

```diff
@@ -423,6 +423,9 @@ export default class Web3Service extends UnlockService {
     ]).then(async ([blockNumber, blockTransaction]) => {
       if (!blockTransaction && !defaults) {
         // transaction is pending, but we have refreshed the page
+        // even though the transaction may not exist, we poll for it
+        // in case it is soon to exist
+        this._watchTransaction(transactionHash)
```

Comment:
It may be due to me missing surrounding context, but would this poll or would this attempt to fetch the transaction again during the next block period?

Reply:
it will attempt to fetch again, and if the transaction turns out to be in a pending state, it will poll until it is fully confirmed. The logic is spread out amongst several functions